hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/BatchConsumer.java

```diff
@@ -60,31 +67,38 @@ public class BatchConsumer implements Consumer {
     private final BatchConsumerMetrics metrics;
     private MessageBatchReceiver receiver;
 
+    private Instant lastCommitTime;
+
```

Comment:
A couple of issues about how `BatchConsumer` uses `OffsetsSlots`:
- we never call `tryToAcquireSlot` which means that `offsetQueueSize` doest not really matter and the set of offsets to be committed is unbounded
- `markAsSent` calls `inflightSemaphore.release()`, but we never acquire the semaphore so number of permits just grows indefinitely  
- `commitIfReady()` calls `offsetsSlots.offsetSnapshot()` which releases the semaphore guarding the slots map. But we never acquire the semaphore so number of permits just grows indefinitely